docs/modules/ROOT/pages/how-tos/configuring/customizing-the-build.adoc

```diff
@@ -176,6 +176,103 @@ An example of a custom task added to the pipeline that sends a slack notificatio
 
 Custom Tasks may need access to data from other Tasks. However, in order to not break the chain of trust in a build Pipeline, there are restrictions in modifying such data. For example, a custom Task should not be allowed to modify the component's source code. If you are using the Enterprise Contract (EC) to verify your builds, introducing a custom Task may violate the link:https://enterprisecontract.dev/docs/ec-policies/release_policy.html#trusted_task__trusted[Trusted Tasks] rule. See xref:advanced-how-tos/using-trusted-artifacts.adoc[Trusted Artifacts] for how to safely allow share data between Tasks.
 
+== Extending the build pipeline with your own tasks using Trusted Artifacts
+
+If you are using the Enterprise Contract (EC) to verify your builds, you can extend the build pipeline with your own tasks using Trusted Artifacts. Trusted Artifacts allow you to share data between Tasks in a secure way. For example, the `task-git-clone` task, has Trusted Artifacts version called `task-git-clone-oci-ta`. 
+[NOTE]
+====
+You can find more tasks link:https://quay.io/organization/konflux-ci[here]. Task generating Trusted Artifacts have `-oci-ta` suffix.
+====
+
+.Adding oci-ta task to pipeline 
+
+In your Pipeline definition, add a new task to the `tasks` section.
+
+[source,yaml]
+--
+spec:
+  tasks:
+    # ...
+    - name: clone-repository-oci-ta
+      params:
+      - name: url
+        value: $(params.git-url)
+      - name: revision
+        value: $(params.revision)
+      - name: ociStorage
+        # needs to be unique storage name
+        value: $(params.output-image).git
+      runAfter:
+      - init
+      taskRef:
+        params:
+        - name: name
+          value: git-clone-oci-ta
+        - name: bundle
+          value: quay.io/konflux-ci/tekton-catalog/task-git-clone-oci-ta@sha256:0f4360ce144d46171ebd2e8f4d4575539a0600e02208ba5fc9beeb2c27ddfd4c
+        - name: kind
+          value: task
+        resolver: bundles
+      workspaces:
+      # use the git-auth workspace for credentials
+      - name: basic-auth
+        workspace: git-auth  
```

Comment:
This looks like the definition for the default git-clone task. Are you suggesting that users add this or are you just using this as an example to show the different parts? If the latter, I think we should use a different pattern.

Reply:
If we want to use the TA for custom tasks to comply with the EC, we have to use the `quay.io/konflux-ci/tekton-catalog/task-git-clone-oci-ta`. We had to add it as a new task.

It is practically the same as the default task. But the default task uses workspaces which means custom tasks will breach the EC.